openassessment/workflow/models.py

```diff
@@ -287,6 +287,17 @@ def update_from_assessments(self, assessment_requirements):
 
         step_for_name = {step.name: step for step in steps}
 
+        # If the status is done or cancelled, check if score has changed.
+        if self.status == self.STATUS.done:
+            if force_update_score:
+                new_score = self.get_score(assessment_requirements, step_for_name)
+                self.set_score(new_score)
+                self.save()
+                logger.info((
+                    u"Workflow for submission UUID {uuid} has updated score."
+                ).format(uuid=self.submission_uuid))
+            return
```

Comment:
When a staff assessment is given, we pass `force_update_score = True` to this method. If status is not `done` the normal flow continues but if status is `done`, the score is updated.

Is there a better way of updating the score after the workflow is `done`?